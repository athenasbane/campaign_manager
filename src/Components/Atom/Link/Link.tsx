import Typography, { TypographyProps } from "@mui/material/Typography";
import { TTypographyVariant } from "Types/Types/link.type";
import { Link as RouterLink } from "react-router-dom";
import theme from "theme";

export interface ILinkProps {
  typographyVariant: TTypographyVariant;
  path: string;
  linkDisplayLabel: string;
  typographyComponentProps?: TypographyProps;
  display?: string;
  color?: string;
}

export default function Link({
  typographyVariant,
  path,
  linkDisplayLabel,
  typographyComponentProps = {},
  display,
  color = theme.palette.contrastText,
}: ILinkProps) {
  return (
    <RouterLink style={{ textDecoration: "none", color: "inherit" }} to={path}>
      <Typography
        data-testid="link_typography"
        color={color}
        sx={{ display: display || "" }}
        {...typographyComponentProps}
        variant={typographyVariant}
      >
        {linkDisplayLabel}
      </Typography>
    </RouterLink>
  );
}
