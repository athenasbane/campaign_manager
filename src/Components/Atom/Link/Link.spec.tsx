import { render, screen } from "@testing-library/react";
import Link, { ILinkProps } from "Components/Atom/Link/Link";
import { BrowserRouter } from "react-router-dom";

const genericProps: ILinkProps = {
  typographyVariant: "h1",
  route: "/test/route",
  linkDisplayLabel: "Testable Link Text",
};

describe("Atom - Link", () => {
  describe("generic prop usage", () => {
    it("should display the correct text", () => {
      render(<Link {...genericProps} />, {
        wrapper: BrowserRouter,
      });
      expect(screen.getByText(/Testable Link Text/i)).toBeInTheDocument();
    });
  });
  describe("Snapshot - Test", () => {
    it("should match snapshot", () => {
      const { container } = render(<Link {...genericProps} />, {
        wrapper: BrowserRouter,
      });
      expect(container).toMatchSnapshot();
    });
  });
});
