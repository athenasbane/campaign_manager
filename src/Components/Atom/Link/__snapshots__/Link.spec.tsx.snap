// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Atom - Link Snapshot - Test should match snapshot 1`] = `
<div>
  <a
    href="/test/route"
    style="text-decoration: none;"
  >
    <h1
      class="MuiTypography-root MuiTypography-h1 css-o2w69a-MuiTypography-root"
      data-testid="link_typography"
    >
      Testable Link Text
    </h1>
  </a>
</div>
`;
